### PR TITLE
run stress test in mini mode with tinylicious driver

### LINKED ISSUE #5640: Service Load Test:  Run in Buddy to protect the test from regression
## Work Item
Add support to the service load test for local server. To support this the orchestrator(nodeStressTest) will need to support launching the running in-process. The runner will need to support being run in memory, just split main into main and run, run takes arguments, and returns, main handles getting args, calling run, and exiting the process. We should then add a step into the npm:test command to run a small version of stress. this is primariyly to make sure the stress test itself isn't broken.

### PR DESCRIPTION
**Pull request by chensixx**
fixes #5640

### CODE REVIEW

File: packages/test/test-service-load/package.json [hunk 1]
```diff
@@ -28,6 +28,7 @@
     "lint:fix": "npm run eslint:fix",
     "start": "node ./dist/nodeStressTest.js",
     "start:mini": "node ./dist/nodeStressTest.js --profile mini",
+    "start:hi": "node ./dist/mini.js",
     "tsc": "tsc",
```
Comment (anthony-murphy) [inline packages/test/test-service-load/package.json:32]: i think you need to add mocha dev dependecies here

File: packages/test/test-service-load/src/test/mini.spec.ts [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+ * Licensed under the MIT License.
+ */
+
+import child_process from "child_process";
+import commander from "commander";
+import { TestDriverTypes } from "@fluidframework/test-driver-definitions";
+import { ILoadTestConfig } from "../testConfigFile";
+import { createTestDriver, getProfile, initialize, safeExit } from "../utils";
+
+async function main() {
```
Comment (anthony-murphy) [inline packages/test/test-service-load/src/test/mini.spec.ts:12]: not sure if this is just for testing, but you shouldn't re-implement anything here. you should just run the current orchestrator via a child process:
https://nodejs.org/api/child_process.html

File: packages/test/test-service-load/package.json (hunk 1 above)
Comment (chensixx) [reply packages/test/test-service-load/package.json:32]: "@types/mocha": "^8.2.2", this one is already in dev dependency. is there another one?

File: packages/test/test-service-load/src/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,15 @@
+{
+    "extends": "@fluidframework/build-common/ts-common-config.json",
+    "compilerOptions": {
+        "declaration": false,
+        "declarationMap": false,
+        "rootDir": "./",
+        // "outDir": "../../dist/test",
```
[the remark this answers is not in the document]
Comment (anthony-murphy) [reply packages/test/test-service-load/src/tsconfig.json:7]: this should also be in the test folder with the test

File: packages/test/test-service-load/src/test/mini.spec.ts [hunk 4]
```diff
@@ -0,0 +1,120 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import child_process from "child_process";
```
Comment (anthony-murphy) [inline packages/test/test-service-load/src/test/mini.spec.ts:6]: also put this a test folder

File: packages/test/test-service-load/package.json [hunk 5]
```diff
@@ -27,7 +27,7 @@
     "lint": "npm run eslint",
     "lint:fix": "npm run eslint:fix",
     "start": "node ./dist/nodeStressTest.js",
-    "start:mini": "node ./dist/nodeStressTest.js --profile mini",
+    "start:mini": "node ./dist/nodeStressTest.js --driver local --profile mini",
```
Comment (anthony-murphy) [inline packages/test/test-service-load/package.json:30]: local will not work here. it should be tinylicious

File: packages/test/test-service-load/package.json [hunk 6]
```diff
@@ -74,6 +74,7 @@
     "@fluidframework/test-utils": "^0.40.0",
     "@fluidframework/tool-utils": "^0.40.0",
     "commander": "^5.1.0",
+    "mocha": "^8.4.0",
```
Comment (anthony-murphy) [inline packages/test/test-service-load/package.json:77]: this should go in dev dependencies

File: packages/test/test-service-load/package.json (hunk 5 above)
Comment (chensixx) [reply packages/test/test-service-load/package.json:30]: oops it was just for testing

File: packages/test/test-service-load/package.json [hunk 7]
```diff
@@ -27,7 +28,12 @@
     "lint": "npm run eslint",
     "lint:fix": "npm run eslint:fix",
     "start": "node ./dist/nodeStressTest.js",
-    "start:mini": "node ./dist/nodeStressTest.js --profile mini",
+    "start:mini": "node ./dist/nodeStressTest.js --driver tinylicious --profile mini",
```
Comment (anthony-murphy) [inline packages/test/test-service-load/package.json:31]:
```suggestion
    "start:mini": "node ./dist/nodeStressTest.js  --profile mini",
```
let's not update the default mode for the mini target

File: packages/test/test-service-load/package.json [hunk 8]
```diff
@@ -74,7 +80,9 @@
     "@fluidframework/test-utils": "^0.40.0",
     "@fluidframework/tool-utils": "^0.40.0",
     "commander": "^5.1.0",
-    "random-js": "^1.0.8"
+    "cross-env": "^7.0.2",
```
Comment (anthony-murphy) [inline packages/test/test-service-load/package.json:83]: the new dependencies should be devDepenedencies

File: packages/test/test-service-load/package.json [hunk 9]
```diff
@@ -27,7 +28,12 @@
     "lint": "npm run eslint",
     "lint:fix": "npm run eslint:fix",
     "start": "node ./dist/nodeStressTest.js",
-    "start:mini": "node ./dist/nodeStressTest.js --profile mini",
+    "start:mini": "node ./dist/nodeStressTest.js --driver tinylicious --profile mini",
+    "start:tinylicious:test": "tinylicious > tinylicious.log 2>&1",
+    "test": "npm run test:realsvc",
```
Comment (anthony-murphy) [inline packages/test/test-service-load/package.json:33]: i don;t think this works. also i don't see the test in the pr results if i search for the title "should return 0"

Comment (chensixx) [reply packages/test/test-service-load/package.json:33]: i see. I just copied from the e2e package.json. which script actually triggers the CI?

Comment (anthony-murphy) [reply packages/test/test-service-load/package.json]: start at the root package.json, and look at which npm scripts ci:test calls, and ensure that this package has a npm script that will be called along that path

Comment (chensixx) [reply packages/test/test-service-load/package.json]: oh! I forgot to update. I found it and it was `: report`, I added it and the test is in the pipeline now!
thanks for the help from Helio and Arin

**Review by anthony-murphy — APPROVED** [review]

**Pull request merged**